ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/TransactionServiceGetEventsIT.scala

```diff
@@ -0,0 +1,355 @@
+      )
+      hasKey1 <- hasKey(beginExclusive = None, endInclusive = None)
+      hasKey2 <- hasKey(
+        beginExclusive =
+          Some(LedgerOffset(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_BEGIN))),
+        endInclusive =
+          Some(LedgerOffset(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_END))),
+      )
+      hasKey3 <- hasKey(beginExclusive = None, endInclusive = offset2)
+      hasKey4 <- hasKey(beginExclusive = None, endInclusive = offset1)
+      hasKey5 <- hasKey(beginExclusive = offset1, endInclusive = None)
+      hasKey6 <- hasKey(beginExclusive = offset2, endInclusive = None)
+      hasKey7 <- hasKey(beginExclusive = offset2, endInclusive = offset2)
+    } yield {
+      assert(hasKey1, s"Unspecified offsets do not restrict selection")
+      assert(hasKey2, s"Ledger boundaries do not restrict selection")
+      assert(hasKey3, s"End offset on key is included")
+      assert(!hasKey4, s"End offset before key is excluded")
+      assert(hasKey5, s"Begin offset before key is included")
+      assert(!hasKey6, s"Begin offset on key is excluded")
+      assert(!hasKey7, s"Identical begin/end offset results in no rows")
+    }
+  })
+
+  test(
+    "TXEventsByContractKeyPaging",
+    "Should page requests if there may be more rows",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    val exercisedKey = "paging key"
+    val key = makeTextKeyKey(party, exercisedKey)
+
+    def getNextTwoResults(
+        beginExclusive: LedgerOffset
+    ): Future[Option[LedgerOffset]] = {
+      ledger
+        .getEventsByContractKey(
+          GetEventsByContractKeyRequest(
+            contractKey = Some(key),
+            templateId = Some(TextKey.id.unwrap),
+            requestingParties = Tag.unsubst(immutable.Seq(party)),
+            maxEvents = 2,
+            beginExclusive = Some(beginExclusive),
+            endInclusive = None,
+          )
+        )
+        .map(_.lastOffset)
+    }
+
+    for {
+      textKeyCid1 <- ledger.create(party, TextKey(party, exercisedKey, Nil))
+      _ <- ledger.submitAndWaitForTransactionTree(
+        ledger.submitAndWaitRequest(party, textKeyCid1.exerciseTextKeyChoice().command)
+      )
+      textKeyCid2 <- ledger.create(party, TextKey(party, exercisedKey, Nil))
+      _ <- ledger.submitAndWaitForTransactionTree(
+        ledger.submitAndWaitRequest(party, textKeyCid2.exerciseTextKeyChoice().command)
+      )
+      lastOffset1 <- getNextTwoResults(
+        LedgerOffset(LedgerOffset.Value.Boundary(LedgerOffset.LedgerBoundary.LEDGER_BEGIN))
+      )
+      lastOffset2 <- getNextTwoResults(assertDefined(lastOffset1, "Expected lastOffset1"))
+      lastOffset3 <- getNextTwoResults(assertDefined(lastOffset2, "Expected lastOffset2"))
+    } yield {
+      assertEquals("Expected the final offset to be empty", lastOffset3, None)
```

Comment:
You also need to extend the `ParticiantPruningIT`

Reply:
The `ParticipantPruningIT` will be exteneded to show how when reporting over a period that has been pruned:
- Events associated with active contracts are still present
- Events associated with consumed contracts are not